swap/swap_test.go

```diff
@@ -57,6 +57,8 @@ var (
 	testChequeSig      = common.Hex2Bytes("fd3f73c7a708bb4e42471b76dabee2a0c1b9af29efb7eadb37f206bf871b81cf0c7987ad89633be930a63eba9e793cc77896131de7d9740b49da80c23c217c621c")
 	testChequeContract = common.HexToAddress("0x4405415b2B8c9F9aA83E151637B8378dD3bcfEDD")
 	gasLimit           = uint64(8000000)
+	testBackend        *backends.SimulatedBackend
+	errc               chan error
```

Comment:
This channel is just closed, it should be `chan struct{}`, and probably should be renamed to something like submitDone.

I would also like to see that this channel is not in the global scope if that is even possible. Not crucial.